src/main/java/com/devonfw/ide/sonarqube/common/impl/check/namingconvention/DevonNamingConventionClassExtendsClassCheck.java

```diff
@@ -0,0 +1,189 @@
+import java.util.regex.Pattern;
+
+import org.sonar.java.ast.parser.JavaParser;
+import org.sonar.java.model.JavaTree;
+import org.sonar.java.model.ModifiersUtils;
+import org.sonar.plugins.java.api.JavaFileScanner;
+import org.sonar.plugins.java.api.JavaFileScannerContext;
+import org.sonar.plugins.java.api.tree.ClassTree;
+import org.sonar.plugins.java.api.tree.CompilationUnitTree;
+import org.sonar.plugins.java.api.tree.IdentifierTree;
+import org.sonar.plugins.java.api.tree.ListTree;
+import org.sonar.plugins.java.api.tree.Modifier;
+import org.sonar.plugins.java.api.tree.Tree;
+import org.sonar.plugins.java.api.tree.Tree.Kind;
+import org.sonar.plugins.java.api.tree.TypeTree;
+
+import com.sonar.sslr.api.typed.ActionParser;
+
+public abstract class DevonNamingConventionClassExtendsClassCheck implements JavaFileScanner {
+
+  private JavaFileScannerContext context;
+
+  final protected String extendedSuperClass;
+
+  final protected String classSuffixRegEx;
+
+  final protected Boolean isAbstract;
+
+  protected String className;
+
+  protected List<String> interfacesToImplement = null;
+
+  public DevonNamingConventionClassExtendsClassCheck(String pExtendedSuperClass, String pClassSuffixRegEx,
+      Boolean pIsAbstract) {
+
+    this.extendedSuperClass = pExtendedSuperClass;
+    this.classSuffixRegEx = pClassSuffixRegEx;
+    this.isAbstract = pIsAbstract;
+  }
+
+  public DevonNamingConventionClassExtendsClassCheck(String pExtendedSuperClass, String pClassSuffixRegEx) {
+
+    this.extendedSuperClass = pExtendedSuperClass;
+    this.classSuffixRegEx = pClassSuffixRegEx;
+    this.isAbstract = null;
+  }
+
+  @Override
+  public void scanFile(JavaFileScannerContext context) {
+
+    this.context = context;
+
+    File file = context.getFile();
+
+    CompilationUnitTree parsedTree = parseJavaFile(file.getPath());
+
+    List<Tree> types = parsedTree.types();
+    ClassTree tree = getTreeInstance(types);
+    Kind kind = tree.kind();
+
+    boolean isClass = kind.equals(Tree.Kind.CLASS);
+    if (!isClass)
+      return;
+
+    TypeTree superClass = tree.superClass();
+
+    LinkedList<String> superInterfacesNames = new LinkedList<String>();
+    ListTree<TypeTree> superInterfaces = tree.superInterfaces();
+
+    IdentifierTree simpleName = tree.simpleName();
+    String className = simpleName.name();
+
+    this.className = className;
+
+    init();
+
+    for (Iterator<TypeTree> iterator = superInterfaces.iterator(); iterator.hasNext();) {
+      TypeTree typeTree = iterator.next();
+      superInterfacesNames.add(typeTree.toString());
+    }
+
+    String superClassName = null;
+    if (superClass != null) {
+      superClassName = superClass.toString();
+    }
+
+    if (superClassName != null) {
+
+      if (superClassName.equals(this.extendedSuperClass)) {
+
+        Pattern pattern = Pattern.compile(this.classSuffixRegEx);
+        Matcher matcher = pattern.matcher(className);
+        boolean endsWith = matcher.find();
```

Comment:
Since the regex is always given to this class with a '$' at the end, I don't see a problem here. I have tested it a few times and so far there was no case where an issue came up. Should I still try to implement it in a different way?

Reply:
> super("AbstractUc", "Uc.*Impl$", false);

This will also match `BananaUcFooImpl` just to give some stupid example. IMHO it is just kind of odd to use `find` here where actually `match` is the correct usage. The `find` method is useful to find sub-strings and especially to do multiple replacements.

Reply:
That's a good point... I'll think of a different solution.